docs/configuration.md

```diff
@@ -83,20 +222,33 @@ all chains. For example the OffRamp is only available to nodes configured with
 an OffRamp reader role. See the [consensus documentation](consensus.md) for
 details about participant roles.
 
-TODO: more details about discovery and a link to the discovery processor.
+Contract discovery is managed by the [discovery processor](https://github.com/smartcontractkit/chainlink-ccip/tree/b99bec3d0692a2224aaa477a7aa297f0b48eaa58/internal/plugincommon/discovery),
+a shared component used by the commit and execute plugins.
 
 ```mermaid
 sequenceDiagram
 JobSpec ->> DestReader: Get HomeChain and OffRamp addresses.
 HomeChain ->> DestReader: Get configuration.
-DestChain(OffRamp) ->> DestReader: Get static configuration.
-DestChain(OffRamp) ->> DestReader: Get dynamic configuration.
-DestChain(OffRamp) ->> DestReader: Get source chain configuration.
+OffRamp ->> DestReader: Get static configuration.
+OffRamp ->> DestReader: Get dynamic configuration.
+OffRamp ->> DestReader: Get source chain configuration.
 DestReader ->> DON: Share contract addresses.
-DON ->> SourceReader(s): Receive contract addresses.
-SourceChain(OnRamp) ->> SourceReader(s): Receive OnRamp configuration
+OnRamp ->> SourceReader(s): Get static configuration.
+OnRamp ->> SourceReader(s): Get dynamic configuration.
+OnRamp ->> SourceReader(s): Get source chain configuration.
+SourceReader(s) ->> DON: Share contract addresses.
 ```
```

Comment:
I think this mermaid diagram is an excellent resource but I wonder if the arrow directions should be reversed - I'm understanding the arrows here basically as abstract "function calls", e.g

```mermaid
sequenceDiagram
OnRamp ->> SourceReader(s): Get static configuration.
```

I'm reading this as "OnRamp calls SourceReader to get the static configuration" but I feel like it should be the other way around:

```mermaid
sequenceDiagram
SourceReader(s) ->> OnRamp(s): Get static configuration.
```

Reply:
Improved this with notes and the full call/response

![image](https://github.com/user-attachments/assets/2b8ecd88-e7be-4397-bdc0-7acb4b7a6347)